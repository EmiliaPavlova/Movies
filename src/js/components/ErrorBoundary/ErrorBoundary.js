import React, { Component, Fragment } from 'react';

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      error: null,
      errorInfo: null
    };
  }

  componentDidCatch(error, info) {
    this.setState({
      hasError: true,
      error: error,
      errorInfo: info
    });
  }

  render() {
    if (this.state.hasError) {
      return (
        <Fragment>
          <h1>Something went wrong.</h1>
          <details style={{ whiteSpace: "pre-wrap" }}>
            {this.state.error && this.state.error.toString()}
            {this.state.errorInfo.componentStack}
          </details>
        </Fragment>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;