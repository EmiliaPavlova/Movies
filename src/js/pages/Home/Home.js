import React, { Component, Fragment } from 'react';
import Header from '../../containers/Header/Header';
import InfoLine from '../../components/InfoLine/InfoLine';
import Results from '../../containers/Results/Results';
import Footer from '../../containers/Footer/Footer';
import './Home.css';

class Home extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <InfoLine />
        <Results />
        <Footer />
      </Fragment>
    );
  }
}

export default Home;
