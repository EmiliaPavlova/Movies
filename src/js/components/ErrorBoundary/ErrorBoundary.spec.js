import React from 'react';
import { mount } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';
import Home from '../../pages/Home/Home';

const componentProps = (overrides = {}) => ({
  children: <div>Component with an error</div>,
  ...overrides
});

const mountErrorBoundary = props => mount(<ErrorBoundary {...props} />);

const ProblemChild = () => {
  throw new Error('Error thrown from problem child');
  return <div>Error</div>;
}

describe('ErrorBoundary', () => {
  describe('ErrorBoundary attributes', () => {
    it('matches the snapshot', () => {
      const wrapper = mountErrorBoundary(componentProps());
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Catching an error', () => {
    it('should display an error when error in state', () => {
      const wrapper = mountErrorBoundary(componentProps());
      wrapper.setState({
        hasError: true,
        error: 'Error',
        errorInfo: { componentStack: 'some additional info' }
      });
      wrapper.update();
      expect(wrapper.html()).toContain('Something went wrong');
    });
    it('should call componentDidCatch', () => {
      const didMountSpy = jest.spyOn(ErrorBoundary.prototype, 'componentDidCatch');
      // mountErrorBoundary(componentProps());
      mount(<ErrorBoundary><ProblemChild /></ErrorBoundary>);

      expect(didMountSpy).toHaveBeenCalledTimes(1);

      didMountSpy.mockClear();
    });
  })
});
