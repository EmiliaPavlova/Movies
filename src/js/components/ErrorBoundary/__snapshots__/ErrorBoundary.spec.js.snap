// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary ErrorBoundary attributes matches the snapshot 1`] = `
<ErrorBoundary>
  <div>
    Component with an error
  </div>
</ErrorBoundary>
`;
